packages/panzer/disc-fe/src/lof/Panzer_BlockedEpetraLinearObjFactory_impl.hpp

```diff
@@ -1233,34 +1367,73 @@ buildColMap(int i) const
    return Teuchos::rcp(new Epetra_Map(-1,indices.size(),&indices[0],0,*eComm_));
 }
 
-// build the ghosted map
-template <typename Traits,typename LocalOrdinalT>
-const Teuchos::RCP<Epetra_Map> BlockedEpetraLinearObjFactory<Traits,LocalOrdinalT>::
+///////////////////////////////////////////////////////////////////////////////
+//
+//  buildGhostedMap()
+//
+///////////////////////////////////////////////////////////////////////////////
+template <typename Traits, typename LocalOrdinalT>
+const Teuchos::RCP<Epetra_Map>
+BlockedEpetraLinearObjFactory<Traits, LocalOrdinalT>::
 buildGhostedMap(int i) const
 {
-   std::vector<int> indices;
-
-   // get the global indices
-   getGlobalIndexer(i)->getOwnedAndGhostedIndices(indices);
+  std::vector<int> indices;
+  getGlobalIndexer(i)->getOwnedAndGhostedIndices(indices);
+  return Teuchos::rcp(new Epetra_Map(-1, indices.size(), &indices[0], 0,
+    *eComm_));
+} // end of buildGhostedMap()
 
-   return Teuchos::rcp(new Epetra_Map(-1,indices.size(),&indices[0],0,*eComm_));
-}
+///////////////////////////////////////////////////////////////////////////////
+//
+//  buildGhostedMap2()
+//
+///////////////////////////////////////////////////////////////////////////////
+template <typename Traits, typename LocalOrdinalT>
+const Teuchos::RCP<Epetra_Map>
+BlockedEpetraLinearObjFactory<Traits, LocalOrdinalT>::
+buildGhostedMap2(int i) const
+{
+  std::vector<int> indices;
+  getGlobalIndexer(i)->getGhostedIndices(indices);
+  return Teuchos::rcp(new Epetra_Map(-1, indices.size(), &indices[0], 0,
+    *eComm_));
+} // end of buildGhostedMap2()
```

Comment:
Same as original, but calls `getGhostedIndices` instead of `getOwnedAndGhostedIndices`.